Develop2

vedmed117 opened the pull request:
Added every necessary test + Allure annotations

vedmed117 committed:
Added pom.xml
Added User.java

vedmed117 committed:
Added /auth/register tests

vedmed117 committed:
Added parametrization into createUserWithMissingFieldTest

vedmed117 committed:
Added orders-pojo classes OrderRequest and OrderResponse

vedmed117 committed:
Added OrderCreationTest

vedmed117 committed:
fixed OrderCreationTest

vedmed117 committed:
CreateUserWithMissingFieldTest created
Added annotations for Allure-report

pomegranate282922 commented on src/main/java/org/praktikum/orders/OrderRequest.java:
```diff
@@ -0,0 +1,19 @@
+
+    public void setIngredients(List<String> ingredients) {
+        this.ingredients = ingredients;
+    }
```
Можно использовать библиотеку Lombok для уменьшения числа boilerplate кода 
Можно улучшить

pomegranate282922 commented on src/test/java/AuthLoginTest.java:
```diff
@@ -0,0 +1,101 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://stellarburgers.nomoreparties.site/api";
```
Все урлы выносим как константы 
Base uri присваиваем один раз в проекте 
Нужно исправить

pomegranate282922 commented on src/test/java/AuthLoginTest.java:
```diff
@@ -0,0 +1,101 @@
+        return new UserRequest(email, password, name);
+    }
+
+    @Step("Register a new user")
```
Вызовы ручек нужно вынести в отдельные классы 
Нужно исправить

pomegranate282922 commented on src/test/java/AuthLoginTest.java:
```diff
@@ -0,0 +1,101 @@
+                .header("Content-type", "application/json")
+                .body(user)
+                .when()
+                .post("/auth/register");
```
Все урлы выносим как константы 
Нужно исправить

pomegranate282922 commented on src/test/java/AuthUserUpdateTest.java:
```diff
@@ -0,0 +1,113 @@
+    @Description("This test verifies that a user can update their data when authenticated.")
+    public void updateUserWithAuthorization() {
+        UserRequest updatedRequest = createUserData(
+                "updated-email_" + System.currentTimeMillis() + "@yandex.ru",
```
Отлично что есть рандомизация. 
удобно использовать библиотеку faker 
Можно улучшить

pomegranate282922 commented on src/test/java/OrderCreationTest.java:
```diff
@@ -0,0 +1,135 @@
+    @DisplayName("Create order with authorization")
+    @Description("This test verifies the creation of an order with authorization")
+    public void createOrderWithAuthTest() {
+        OrderRequest orderRequest = createOrder(Arrays.asList("61c0c5a71d1f82001bdaaa6d"));
```
Ингредиенты не хардкодим, а берем из соответствующей ручки 
Нужно исправить

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Есть рандомизация. Везде есть удаление данных.
Замечания:
1. Base uri присваиваем один раз в проекте 
2. Все вызовы ручек выносим из тестовых классов 
3. Все урлы выносим как константы 
4. Ингредиенты не хардкодим 
5. Нет аллюр отчета

vedmed117 committed:
OrderRequest - added const-variables
ApiSteps - added new class with ALL steps, used in test-classes
BASE_URI - added new class with Base URI
Classes with tests - refactored, all reoccurring methods were moved into ApiSteps, instead of using URLs and consts now uses methods from classes with data

vedmed117 committed:
Added Allure-results

pomegranate282922 commented on src/main/java/org/praktikum/utils/BASE_URI.java:
```diff
@@ -0,0 +1,10 @@
+
+    private static final String BASE_URI = "https://stellarburgers.nomoreparties.site/";
+
+    public static String getBaseURI() {
```
Можно просто публичную константу сделать

pomegranate282922 commented on src/test/java/AuthRegisterTest.java:
```diff
@@ -0,0 +1,64 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = BASE_URI.getBaseURI();
```
Base uri присваиваем один раз в проекте 
Нужно исправить

pomegranate282922 reviewed (commented):
Многое исправил - супер!
Остались замечания:
1. base uri нужно присваивать один (!) раз в проекте 
2. Также отсутствует тест на получение заказов - прошу добавить

vedmed117 committed:
Added RestAssuredConfig.java with BaseURI initialization, now other classes are using this
Added OrderRetrievalTest with two tests: getting order as authorized and non-authorized user

vedmed117 committed:
allure results

pomegranate282922 commented on src/main/java/org/praktikum/utils/API_URI.java:
```diff
@@ -0,0 +1,30 @@
+package org.praktikum.utils;
+
+public class API_URI {
```
CamelCase

pomegranate282922 commented on src/main/java/org/praktikum/utils/BASE_URI.java:
```diff
@@ -0,0 +1,16 @@
+
+import io.restassured.RestAssured;
+
+public class BASE_URI {
```
CamelCase

pomegranate282922 commented on src/test/java/AuthLoginTest.java:
```diff
@@ -0,0 +1,54 @@
+    @Before
+    public void setUp() {
+        RestAssuredConfig.init();
+        RestAssured.baseURI = BASE_URI.getBaseURI();
```
А в чем смысл тогда дважды присваивать?

pomegranate282922 reviewed (commented):
Есть еще парочка замечаний по неймингу, но будем считать некритичным - зачет!
